plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergModule.java

```diff
@@ -96,4 +102,11 @@ public void configure(Binder binder)
         tableProcedures.addBinding().toProvider(ExpireSnapshotsTableProcedure.class).in(Scopes.SINGLETON);
         tableProcedures.addBinding().toProvider(RemoveOrphanFilesTableProcedure.class).in(Scopes.SINGLETON);
     }
+
+    @Singleton
+    @Provides
+    public ExecutorService createIcebergExecutor()
+    {
+        return newCachedThreadPool(daemonThreadsNamed("iceberg-executor-%s"));
```

Comment:
Who shuts this down on connector shutdown?

Reply:
Good catch. I think Delta is missing the shutdown step as well. I've changed the code to inject the iceberg executor service to the IcebergConnector so that it can be shut down when `IcebergConnector.shutDown()` is called